content/en/platform/corda/5.0/developing-applications/api/api-crypto.md

```diff
@@ -101,4 +97,25 @@ dependencies {
 You must reference `net.corda:corda-crypto-extensions`.
 {{< /note >}}
 
-<!--For information about packaging a CorDapp, see the [development tutorial]().-->
\ No newline at end of file
+## Implementing Signature Schemes
+
+Corda supports the following `SignatureSpecs` (signature schemes) for creating `java.security.Signature` objects:
+* SHA256withRSA
+* SHA384withRSA
+* SHA512withRSA
+* RSASSA-PSS with SHA256
+* RSASSA-PSS with SHA384
+* RSASSA-PSS with SHA512
+* RSASSA-PSS with SHA256 and MGF1
+* RSASSA-PSS with SHA384 and MGF1
+* RSASSA-PSS with SHA512 and MGF1
+* SHA256withECDSA
+* SHA384withECDSA
+* SHA512withECDSA
+* EdDSA
+* SHA512withSPHINCS256
+* SM3withSM2
+* SHA256withSM2
+* GOST3411withGOST3410
+
+Use `SignatureSpecService` to retrieve the `SignatureSpec`.
```

Comment:
I think we should elaborate a bit here on `SignatureSpecService`  and why we introduced it. 

Initially we had the above list of signature spec list available to users. But then it could be that they could be passing the wrong signature spec for a signing key type (e.g. an RSA key with an `SHA256withECDSA`), which would lead to an error when attempting to generate the signature.

Instead we introduced  `SignatureSpecService` which takes in a key (and a digest algorithm optionally) and returns a default signature spec for those.